### ISSUE TITLE
max_positions not resolved correctly for validation set?

### ISSUE THREAD

**Issue by foldie** [pytorch/fairseq#2998]
## 🐛 Bug

When validating the max_positions doesn't get passed on to the validation batch_iterator. 

### To Reproduce

Fairseq crashes on my machine when I restrict the maximum token length for the training but the validation data contains sentences that are longer. This happens even though I am using the `--skip-invalid-size-inputs-valid-test` flag.

```
fairseq-train \ 
        /path/to/dataset/data.tokenized.src-trg \ 
        --warmup-updates \ 
        4000 \ 
        --lr \ 
        0.00025 \ 
        --required-batch-size-multiple \ 
        4000 \ 
        --warmup-init-lr \ 
        0.000025 \ 
        --optimizer \ 
        adam \ 
        --lr-scheduler \ 
        inverse_sqrt \ 
        --criterion \ 
        cross_entropy \ 
        --clip-norm \ 
        0.1 \ 
        --dropout \ 
        0.1 \ 
        --attention-dropout \ 
        0.1 \ 
        --skip-invalid-size-inputs-valid-test \ 
        --max-tokens \ 
        200 \ 
        --best-checkpoint-metric \ 
        loss \ 
        --maximize-best-checkpoint-metric \ 
        --arch \ 
        transformer_wmt_en_de \ 
        --seed \ 
        42 \ 
        --save-dir \ 
        /path/to/checkpoints/
```

If I run this I get the unexpected error:

```
sentence at index N of size M exceeds max_tokens limit of 200!
```
I think this is because `self.cfg.dataset.max_tokens` is not passed to `utils.resolve_max_positions` at line 
https://github.com/pytorch/fairseq/blob/ba4f54267af5c3f67f1b76a6e804b6ab593d1d39/fairseq/trainer.py#L454 
as it is passed to the same function at the training step at line
https://github.com/pytorch/fairseq/blob/ba4f54267af5c3f67f1b76a6e804b6ab593d1d39/fairseq/trainer.py#L429

### Expected behavior

Samples that are too long are skipped.

### Environment

 - fairseq Version: master
 - PyTorch Version: 1.4.0
 - OS: Linux
 - How you installed fairseq: source
 - Python version: 3.8.5
 - CUDA/cuDNN version: 7.6.5
